src/Potato/Data/Text/Zipper2.hs

```diff
@@ -154,7 +156,10 @@ pageDown pageSize z = undefined
 
 -- | Move the cursor to the beginning of the current logical line (clearing the selection)
 home :: TextZipper -> TextZipper
-home (TextZipper lb b s a la) = undefined
+home (TextZipper lb b [] a la) = TextZipper lb "" [] (b <> a) la
+home (TextZipper lb b (x:[]) a la) = TextZipper lb "" [] (b <> x <> a) la
+--home (TextZipper lb b (x:xs) a la) = case xs of
+home (TextZipper lb b (x:xs) a la) = TextZipper lb "" [] (b <> x) ((init xs) <> [(last xs) <> a] <> la)
```

Comment:
The pattern matching guarantees xs is non-empty so `last` is fine here, but it's still a [partial function](https://wiki.haskell.org/Partial_functions)

The way to encode the pattern matching non-emptyness guarantee of xs being non empty would be to use [NonEmpty](https://hackage.haskell.org/package/base-4.17.0.0/docs/Data-List-NonEmpty.html) which is constructed using the `(:|) :: a -> [a] -> NonEmpty a` operator. Then you could do something like:
```
home (TextZipper lb b (x:(xs:xss)) a la) = TextZipper lb "" [] (b <> x) ((init xs') <> [(last xs') <> a] <> la) where
   xs' = xs :| xss
```

That's why I went through the trouble of writing appendEnd,  but using NonEmpty might actually be better!

Reply:
Note both Prelude and NonEmpty exports init/last so you need to qualify them i.e.

import Data.List.NonEmpty as NE
NE.init/NE.last